### PR TITLE
Add JSDoc annotations to fully support closure-compiler

### PR DESCRIPTION
**Pull request by ChadKillingsworth**
I've been using and enjoying the library. However with closure-compiler, I have to maintain a custom build to add a few JSDoc annotations.

These changes make the library fully compatible with Closure-Compiler ADVANCED mode.

### CODE REVIEW

**Comment by jonashaag** [comment]
Thanks!

How can we make sure Closure Compiler annotations aren't going to break in future releases? Is there some linter/checker that we can integrate into CI?

File: fingerprint2.js [hunk 1]
```diff
@@ -56,7 +60,7 @@
           if (typeof that.options.preprocessor === 'function') {
             componentValue = that.options.preprocessor(pair.key, componentValue)
           }
-          this.data.push({key: pair.key, value: componentValue})
+          keys.data.push({key: pair.key, value: componentValue})
```
Comment (jonashaag) [inline fingerprint2.js]: Ouch, this must have been a merge error. Thanks for fixing this

**Comment by ChadKillingsworth** [comment]
> How can we make sure Closure Compiler annotations aren't going to break in future releases? Is there some linter/checker that we can integrate into CI?

If you are interested, I can PR a small test case that just runs the library through the compiler with all the checks enabled and ensures there are no warnings or errors.

File: fingerprint2.js (hunk 1 above)
Comment (ChadKillingsworth) [reply fingerprint2.js]: Closure-compiler complained loudly about that one.

**Comment by jonashaag** [comment]
That would be awesome!

**Pull request merged**